File: SK210 - Java - EasyLayer/app/src/main/java/com/example/easylayer_217_sk210/MainActivity.java
package com.example.easylayer_217_sk210;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ListView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.example.easylayer_217_sk210.Impressora.Impressora;
import com.example.easylayer_217_sk210.Projeto.Projeto;
import com.example.easylayer_217_sk210.Projeto.ProjetoAdapter;
import com.example.easylayer_217_sk210.Projeto.Sobre;
import com.example.easylayer_217_sk210.Scanner.Scanner;
import com.example.easylayer_217_sk210.tef.Tef;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ArrayList<Projeto> projetos = new ArrayList<Projeto>();
    ListView lvProjetos;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Informações
        @SuppressLint({"MissingInflatedId", "LocalSuppress"})
        Button informacoes = findViewById(R.id.infor);
        informacoes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showDialog();
            }
        });

//Lista das funções
        lvProjetos = findViewById(R.id.lvProjetos);
        projetos.add(new Projeto("Impressão", R.drawable.print ));
        projetos.add(new Projeto("Scanner", R.drawable.barcode));
        projetos.add(new Projeto("Tef", R.drawable.pos));

        //Intent para as activitys das funções
        ProjetoAdapter adapter = new ProjetoAdapter(getBaseContext(), R.layout.listprojetos, projetos);
        lvProjetos.setAdapter(adapter);
        lvProjetos.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Projeto projeto = (Projeto) lvProjetos.getItemAtPosition(i);

                Intent intent = null;
                switch (projeto.getNome()){
                    case "Impressão":
                        intent = new Intent(MainActivity.this, Impressora.class);
                        break;
                    case "Scanner":
                        intent = new Intent(MainActivity.this, Scanner.class);
                        break;
                    case "Tef":
                        intent = new Intent(MainActivity.this, Tef.class);
                        break;
                }
                if(intent != null){
                    startActivity(intent);
                }
            }
        });
    }

    //Função do dialog de informações
    public void showDialog() {
        int mStackLevel = 1;
        mStackLevel++;

        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        Fragment prev = getSupportFragmentManager().findFragmentByTag("dialog");
        if (prev != null) {
            ft.remove(prev);
        }
        ft.addToBackStack(null);

        Sobre newFragment = Sobre.newInstance(mStackLevel);
        newFragment.show(ft, "dialog");
    }
}

File: SK210 - Kotlin/app/src/main/java/com/exampledv/exemplosk210kot/MainActivity.kt
package com.exampledv.exemplosk210kot

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.widget.Button
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import com.exampledv.exemplosk210kot.camera.Camera
import com.exampledv.exemplosk210kot.impressora.Printer
import com.exampledv.exemplosk210kot.scanner.Scanner
import com.exampledv.exemplosk210kot.sensor.SensorActivity
import com.exampledv.exemplosk210kot.sensor.SensorBobina
import com.exampledv.exemplosk210kot.tef.TefActivity

class MainActivity : AppCompatActivity() {

    private lateinit var btnPrinter: Button
    private lateinit var btnScanner: Button
    private lateinit var btnCamera: Button
    private lateinit var btnSensorBobina: Button
    private lateinit var btnSensorPresenca: Button
    private lateinit var btnTef: Button
    private lateinit var infor: Button

    @RequiresApi(Build.VERSION_CODES.R)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Ativa o modo full screen
        window.setDecorFitsSystemWindows(false)
        val controller = ViewCompat.getWindowInsetsController(window.decorView)
        controller?.let {
            it.hide(WindowInsetsCompat.Type.systemBars()) // Oculta as barras de sistema
            it.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE // Permite que as barras apareçam com swipe
        }
        setContentView(R.layout.activity_main)

        // Inicializa os botões
        btnPrinter = findViewById(R.id.btnPrinter)
        btnScanner = findViewById(R.id.btnScanner)
        btnCamera = findViewById(R.id.btnCamera)
        btnSensorBobina = findViewById(R.id.btnSensorBobina)
        btnSensorPresenca = findViewById(R.id.btnSensorPresenca)
        btnTef = findViewById(R.id.btnTef)
        infor = findViewById(R.id.infor)

        // Ação para abrir o diálogo ao clicar no botão "Infor"
        infor.setOnClickListener {
            // Constrói o AlertDialog
            val alertDialog = AlertDialog.Builder(this)
                .setTitle("Informações")
                .setMessage("Equipamento: SK210\nSDK: DecodeLibrary_1.8.03.A28\nLinguagem: Kotlin\nVersão do app: 1.0.0")
                .setPositiveButton("OK") { dialog, _ ->
                    dialog.dismiss() // Fecha o diálogo ao clicar em OK
                }
                .create()

            // Exibe o AlertDialog
            alertDialog.show()
        }

        // Configuração dos outros botões
        btnPrinter.setOnClickListener {
            val intent = Intent(this, Printer::class.java)
            startActivity(intent)
        }

        btnScanner.setOnClickListener {
            val intent = Intent(this, Scanner::class.java)
            startActivity(intent)
        }

        btnCamera.setOnClickListener {
            val intent = Intent(this, Camera::class.java)
            startActivity(intent)
        }

        btnSensorPresenca.setOnClickListener {
            val intent = Intent(this, SensorActivity::class.java)
            startActivity(intent)
        }

        btnSensorBobina.setOnClickListener {
            val intent = Intent(this, SensorBobina::class.java)
            startActivity(intent)
        }

        btnTef.setOnClickListener {
            val intent = Intent(this, TefActivity::class.java)
            startActivity(intent)
        }
    }
}
